python/spider-py/src/spider_py/client/task.py

```diff
@@ -0,0 +1,101 @@
+from collections.abc import Callable
+from types import FunctionType, GenericAlias
+from typing import get_args, get_origin
+
+from spider_py import core
+from spider_py.client.data import Data
+from spider_py.core import TaskInput, TaskOutput, TaskOutputData, TaskOutputValue
+from spider_py.type import to_tdl_type_str
+
+
+class TaskContext:
+    """Spider task context."""
+
+    # TODO: Implement task context for use in task executor
+
+
+# Check the TaskFunction signature at runtime.
+# Enforcing static check for first argument requires the use of Protocol. However, functions, which
+# are Callable, are not considered a Protocol without explicit cast.
+TaskFunction = Callable[..., object]
+
+
+def _is_tuple(t: type | GenericAlias) -> bool:
+    """
+    :param t:
+    :return: Whether t is a tuple.
+    """
+    return get_origin(t) is tuple
+
+
+def _process_parameters(task: core.Task, signature: inspect.Signature) -> None:
+    """
+    Checks the parameters validity and add them to the task.
+    :param task:
+    :param signature:
+    :raises TypeError: If the parameters are invalid.
+    """
+    params = list(signature.parameters.values())
+    if not params or params[0].annotation is not TaskContext:
+        msg = "First argument is not a TaskContext."
+        raise TypeError(msg)
+    for param in params[1:]:
+        if param.kind in {inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD}:
+            msg = "Variadic parameters are not supported."
+            raise TypeError(msg)
+        if param.annotation is inspect.Parameter.empty:
+            msg = "Argument must have type annotation"
+            raise TypeError(msg)
+        tdl_type_str = to_tdl_type_str(param.annotation)
+        task.task_inputs.append(TaskInput(tdl_type_str, None))
+
+
+def _process_return(task: core.Task, signature: inspect.Signature) -> None:
+    """
+    Checks the return type validity and add them to the task.
+    :param task:
+    :param signature:
+    :raises TypeError: If the return type is invalid.
+    """
+    returns = signature.return_annotation
+    if returns is inspect.Parameter.empty:
+        msg = "Return type must have type annotation"
+        raise TypeError(msg)
+    if _is_tuple(returns):
+        args = get_args(returns)
+        if Ellipsis in args:
+            msg = "Variable-length tuple return types are not supported."
+            raise TypeError(msg)
+        for r in args:
+            tdl_type_str = to_tdl_type_str(r)
+            if r is Data:
+                task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputData()))
+            else:
+                task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputValue()))
+    else:
+        tdl_type_str = to_tdl_type_str(returns)
+        if returns is Data:
+            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputData()))
+        else:
+            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputValue()))
```

Comment:
```suggestion

    if not _is_tuple(returns):
        tdl_type_str = to_tdl_type_str(returns)
        if returns is Data:
            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputData()))
        else:
            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputValue()))
        return

    args = get_args(returns)
    if Ellipsis in args:
        msg = "Variable-length tuple return types are not supported."
        raise TypeError(msg)
    for arg in args:
        tdl_type_str = to_tdl_type_str(arg)
        if arg is Data:
            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputData()))
        else:
            task.task_outputs.append(TaskOutput(tdl_type_str, TaskOutputValue()))
```
* Let's try to avoid long if-else cluster. Use early return to simply it so that it's more readable.
* `r` -> `arg` as: (1) we usually don't allow single char variables except looping index like `i`; (2) `r` itself as `return` can be confusing.